homeassistant/components/velbus/__init__.py

```diff
@@ -80,6 +80,31 @@ def syn_clock(self, service=None):
 
     hass.services.async_register(DOMAIN, "sync_clock", syn_clock, schema=vol.Schema({}))
 
+    def set_memo_text(service):
+        """Handle Memo Text service call."""
+        memo_text = service.data[ATTR_MEMO_TEXT]
+        memo_text.hass = hass
+        try:
+            controller.get_module(service.data[ATTR_MODULE_ADDRESS]).set_memo_text(
+                memo_text.async_render()
+            )
+        except velbus.util.VelbusException as err:
+            _LOGGER.error("An error occurred: %s", err)
+
+    hass.services.async_register(
+        DOMAIN,
+        "set_memo_text",
+        set_memo_text,
+        vol.Schema(
+            {
+                vol.Required(ATTR_MODULE_ADDRESS): vol.All(
+                    vol.Coerce(int), vol.Range(min=0, max=255)
+                ),
+                vol.Optional(ATTR_MEMO_TEXT, default=""): cv.template,
```

Comment:
Why not make this required? Or is there a case where an empty message would be useful?
```suggestion
                vol.Required(ATTR_MEMO_TEXT): cv.template,
```

Reply:
There is an use case for an empty text as it will remove any existing memo text to be displayed. This is also mentioned in associated the documenation [PR 11892:](https://github.com/home-assistant/home-assistant.io/pull/11892/files)

Reply:
Okay that is fine.